packages/react-chat/src/contexts/RuntimeContext/index.tsx

```diff
@@ -4,19 +4,22 @@ import { RuntimeState, Settings, useRuntimeState } from './useRuntimeState';
 
 // split up API and state to prevent unnecessary re-renders
 export const RuntimeStateAPIContext = createContext<RuntimeState['api']>({} as any);
-export const RuntimeStateContext = createContext<RuntimeState['state']>({} as any);
+export const RuntimeStateContext = createContext<RuntimeState['state'] & { shadowRoot: ShadowRoot; autostart: boolean }>({} as any);
 
-interface RuntimeProviderProps extends React.PropsWithChildren, Settings {}
+interface RuntimeProviderProps extends React.PropsWithChildren, Settings {
+  shadowRoot: ShadowRoot;
+}
 
-export const RuntimeProvider = ({ children, assistant, config }: RuntimeProviderProps) => {
+export const RuntimeProvider = ({ children, assistant, config, shadowRoot }: RuntimeProviderProps) => {
   const store = useRuntimeState({ assistant, config });
 
   // api is a static object, so we can use useMemo to prevent unnecessary re-renders
   const api = useMemo(() => store.api, []);
+  const state = useMemo(() => ({ ...store.state, shadowRoot, autostart: !!config.autostart }), [config, shadowRoot, store.state]);
```

Comment:
I think we can narrow the scope of dependencies
```suggestion
  const state = useMemo(() => ({ ...store.state, shadowRoot, autostart: !!config.autostart }), [config.autostart, shadowRoot, store.state]);
```